tests/sample/cgroup_sock_addr2.c

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// Whenever this sample program changes, bpf2c_tests will fail unless the
+// expected files in tests\bpf2c_tests\expected are updated. The following
+// script can be used to regenerate the expected files:
+//     generate_expected_bpf2c_output.ps1
+//
+// Usage:
+// .\scripts\generate_expected_bpf2c_output.ps1 <build_output_path>
+// Example:
+// .\scripts\generate_expected_bpf2c_output.ps1 .\x64\Debug\
+
+#include "bpf_helpers.h"
+#include "socket_tests_common.h"
+
+#define IPPROTO_TCP 6
+#define IPPROTO_UDP 17
+
+#define AF_INET 2
+#define AF_INET6 0x17
+
+SEC("maps")
+struct bpf_map_def policy_map = {
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(destination_entry_t),
+    .value_size = sizeof(destination_entry_t),
+    .max_entries = 100};
+
+__inline int
+authorize_v4(bpf_sock_addr_t* ctx)
```

Comment:
Nit: Since these programs redirect connections, please change the function names and also the program names to reflect and emphasize on redirection operation.